src/ucs/arch/x86_64/cpu.h

```diff
@@ -88,15 +89,19 @@ static inline void ucs_arch_clear_cache(void *start, void *end)
 
 static inline void *ucs_memcpy(void *dst, const void *src, size_t len) {
 #if defined(ENABLE_BUILTIN_MEMCPY)
-    asm volatile ("rep movsb"
-                  : "=D" (dst),
-                  "=S" (src),
-                  "=c" (len)
-                  : "0" (dst),
-                  "1" (src),
-                  "2" (len)
-                  : "memory");
-    return dst;
+    if (len <= ucs_global_opts.memcpy_thresh) {
```

Comment:
maybe `likely` to optimize tiny messages path?

Reply:
first impression is "likely" by default